File: src/exemplos/5_EstruturasDeDados/4_Vetor/0-vetor.c
/**      @file: 0-vetor.c
 *     @author: Guilherme N. Ramos (gnramos@unb.br)
 * @disciplina: Algoritmos e Programação de Computadores
 *
 * Exemplo uso de vetor. */

#include <stdio.h>

int main() {
  int vetor[10];
  int i, soma = 0;

  for(i = 0; i < 10; ++i) {
    printf("Digite o %d-ésimo elemento: ", i);
    scanf("%d", vetor+i);
  }

  printf("\nOs elementos são: ");
  for(i = 0; i < 10; ++i)
    printf("%d ", vetor[i]);

  for(i = 9; i >= 0; --i)
    soma += vetor[i];
  printf("\n\nE a soma deles é: %d\n", soma);

  return 0;
}

File: src/exemplos/5_EstruturasDeDados/4_Vetor/0-vetor.py
#  -*- coding: utf-8 -*-
#       @file: 0-vetor.py
#     @author: Guilherme N. Ramos (gnramos@unb.br)
# @disciplina: Algoritmos e Programação de Computadores
#
# Exemplo uso de vetor. Em Python, vetores são implementados como listas.

vetor = [0]*10

for i in range(10):
    vetor[i] = input("Digite o " + str(i) + "-ésimo elemento:")

print "\nOs elementos são:"
print "{%s}" % ' '.join(str(v) for v in vetor)

print "\nE a soma deles é:", sum(vetor)

# soma = 0
# for i in xrange(len(vetor)):
#     soma += vetor[i]
#
# ou então
#
# for valor in vetor:
#     soma += valor
